runtime/service/src/main/java/org/apache/polaris/service/tracing/RequestIdFilter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.tracing;
+
+import jakarta.annotation.Priority;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerRequestFilter;
+import jakarta.ws.rs.container.PreMatching;
+import jakarta.ws.rs.ext.Provider;
+import java.util.UUID;
+import org.apache.polaris.service.config.FilterPriorities;
+import org.apache.polaris.service.logging.LoggingConfiguration;
+import org.slf4j.MDC;
+
+@PreMatching
+@ApplicationScoped
+@Priority(FilterPriorities.REQUEST_ID_FILTER)
+@Provider
+public class RequestIdFilter implements ContainerRequestFilter {
+
+  public static final String REQUEST_ID_KEY = "requestId";
+
+  @Inject LoggingConfiguration loggingConfiguration;
+
+  @Override
+  public void filter(ContainerRequestContext rc) {
+    var requestId = rc.getHeaderString(loggingConfiguration.requestIdHeaderName());
+    if (requestId == null) {
+      requestId = UUID.randomUUID().toString();
+    }
+    MDC.put(REQUEST_ID_KEY, requestId);
```

Comment:
I'm admittedly nit-picking a bit 😅 but I'd argue that the `MDC.put` statement should stay in `LoggingMDCFilter` since it's about logging.

What should be moved here is just the `rc.setProperty` call.